sdk/src/utils.ts

```diff
@@ -0,0 +1,20 @@
+export async function get(url: URL | string) {
+    const response = await fetch(url);
+
+    if (!response.ok) {
+        throw new Error(response.status + " could not get URL " + url);
+    }
+
+    return response;
+}
+
+
+export async function post(url: URL | string, options: RequestInit) {
+    const response = await fetch(url, options);
+
+    if (!response.ok) {
+        throw new Error(response.status + " could not post URL " + url);
+    }
+
+    return response;
+}
```

Comment:
This seems to be failing for the submitTransaction method in the SDK
![image](https://github.com/AleoHQ/sdk/assets/26438809/b0799778-7340-4ede-947f-dbe4513c65df)

Reply:
This is the full trace of execution log statements. Everything else works I just think there's a logical error in this funciton.

```
Executing local function: hello
wasm.js:3027 Check program imports are valid and add them to the process
wasm.js:3027 parsing inputs
wasm.js:3027 Loading program
wasm.js:3027 Loading function
wasm.js:3027 Adding program to the process
wasm.js:3027 Inserting externally provided proving and verifying keys for hello_hello.aleo - hello
wasm.js:3027 Creating authorization
wasm.js:3027 Executing program
worker.js:60 Web worker: Local execution completed in 14625 ms
worker.js:71 Function execution response: 10u32
contentScript.js:1 Intercom connection corrupted A {message: 'Not Found', errors: undefined, name: 'IntercomError'}
worker.js:96 Web worker: Creating execution...
contentScript.js:1 Intercom connection corrupted A {message: 'Not Found', errors: undefined, name: 'IntercomError'}
index.ts:106 Error finding hello_hello.aleo. Network response: 'Error: Error fetching program'. Please ensure you're connected to a valid Aleo network the program is deployed to the network.
f @ index.ts:106
execute @ program-manager.ts:243
await in execute (async)
(anonymous) @ worker.js:122
h @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
l @ worker.js:2
i @ worker.js:2
Promise.then (async)
l @ worker.js:2
i @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:99
worker.js:143 Error creating execution transaction: Error finding hello_hello.aleo. Network response: 'Error: Error fetching program'. Please ensure you're connected to a valid Aleo network the program is deployed to the network.
(anonymous) @ worker.js:143
h @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
l @ worker.js:2
s @ worker.js:2
Promise.then (async)
l @ worker.js:2
i @ worker.js:2
Promise.then (async)
l @ worker.js:2
i @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:99
contentScript.js:1 Intercom connection corrupted A {message: 'Not Found', errors: undefined, name: 'IntercomError'}
contentScript.js:1 Intercom connection corrupted A {message: 'Not Found', errors: undefined, name: 'IntercomError'}
worker.js:293 Web worker: Creating deployment...
worker.js:310 Program not found on the Aleo Network - proceeding with deployment...
worker.js:319 fee is:  1.9
function-key-provider.ts:348 Fetching proving keys from url https://testnet3.parameters.aleo.org/fee_public.prover.5515650
network-client.ts:367 Uncaught (in promise) Error: Error fetching program
    at u.getProgram (network-client.ts:367:13)
getProgram @ network-client.ts:367
await in getProgram (async)
deploy @ program-manager.ts:142
(anonymous) @ worker.js:321
h @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
l @ worker.js:2
s @ worker.js:2
Promise.then (async)
l @ worker.js:2
i @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:296
contentScript.js:1 Intercom connection corrupted A {message: 'Not Found', errors: undefined, name: 'IntercomError'}
utils.ts:2 Fetch finished loading: GET "https://testnet3.parameters.aleo.org/fee_public.prover.5515650".
i @ utils.ts:2
fetchBytes @ function-key-provider.ts:188
fetchKeys @ function-key-provider.ts:349
feePublicKeys @ function-key-provider.ts:432
deploy @ program-manager.ts:174
(anonymous) @ worker.js:321
h @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
l @ worker.js:2
s @ worker.js:2
Promise.then (async)
l @ worker.js:2
i @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:296
function-key-provider.ts:350 Fetching verifying keys fee_public.verifier.40ea40e
wasm.js:3027 Creating deployment transaction
wasm.js:3027 Checking program has a valid name
wasm.js:3027 Checking program imports are valid and add them to the process
wasm.js:3027 Creating deployment
contentScript.js:1 Intercom connection corrupted A {message: 'Not Found', errors: undefined, name: 'IntercomError'}
wasm.js:2892 XHR finished loading: GET "https://s3-us-west-1.amazonaws.com/testnet3.parameters/shifted-powers-of-beta-16.usrs.d99bcb3".
(anonymous) @ wasm.js:2892
S @ wasm.js:315
P._.wbg.__wbg_send_9f5007eae908c72e @ wasm.js:2891
$func100 @ 82ae6b53e2f7420b55e4.wasm:0x40d23
$func1406 @ 82ae6b53e2f7420b55e4.wasm:0x60d6c5
$func2686 @ 82ae6b53e2f7420b55e4.wasm:0x6bbdca
$func110 @ 82ae6b53e2f7420b55e4.wasm:0xe27ff
$func455 @ 82ae6b53e2f7420b55e4.wasm:0x39824f
$func144 @ 82ae6b53e2f7420b55e4.wasm:0x19d09b
$func175 @ 82ae6b53e2f7420b55e4.wasm:0x2185e8
$func1806 @ 82ae6b53e2f7420b55e4.wasm:0x65a8d3
$func2665 @ 82ae6b53e2f7420b55e4.wasm:0x6ba70b
$wasm_bindgen__convert__closures__invoke1_mut__h76225f74e93744a5 @ 82ae6b53e2f7420b55e4.wasm:0x6e5b37
f @ wasm.js:230
o @ wasm.js:215
Promise.then (async)
P._.wbg.__wbg_then_f7e06ee3c11698eb @ wasm.js:3232
$func1485 @ 82ae6b53e2f7420b55e4.wasm:0x61ca5c
$func2977 @ 82ae6b53e2f7420b55e4.wasm:0x6d095b
$wasm_bindgen__convert__closures__invoke2_mut__h0edc44cbd56288e2 @ 82ae6b53e2f7420b55e4.wasm:0x6e5a43
T @ wasm.js:321
(anonymous) @ wasm.js:3072
P._.wbg.__wbg_new_43f1b47c28813cbd @ wasm.js:3077
$programmanager_buildDeploymentTransaction @ 82ae6b53e2f7420b55e4.wasm:0x6d4df4
buildDeploymentTransaction @ wasm.js:1437
deploy @ program-manager.ts:189
await in deploy (async)
(anonymous) @ worker.js:321
h @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
l @ worker.js:2
s @ worker.js:2
Promise.then (async)
l @ worker.js:2
i @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:296
wasm.js:3027 Ensuring the fee is sufficient to pay for the deployment
wasm.js:2892 XHR finished loading: GET "https://s3-us-west-1.amazonaws.com/testnet3.parameters/powers-of-beta-17.usrs.7c27308".
(anonymous) @ wasm.js:2892
S @ wasm.js:315
P._.wbg.__wbg_send_9f5007eae908c72e @ wasm.js:2891
$func99 @ 82ae6b53e2f7420b55e4.wasm:0x2df04
$func100 @ 82ae6b53e2f7420b55e4.wasm:0x2f568
$func1406 @ 82ae6b53e2f7420b55e4.wasm:0x60d6c5
$func2686 @ 82ae6b53e2f7420b55e4.wasm:0x6bbdca
$func110 @ 82ae6b53e2f7420b55e4.wasm:0xe2db9
$func455 @ 82ae6b53e2f7420b55e4.wasm:0x39824f
$func144 @ 82ae6b53e2f7420b55e4.wasm:0x19d09b
$func175 @ 82ae6b53e2f7420b55e4.wasm:0x2185e8
$func1806 @ 82ae6b53e2f7420b55e4.wasm:0x65a8d3
$func2665 @ 82ae6b53e2f7420b55e4.wasm:0x6ba70b
$wasm_bindgen__convert__closures__invoke1_mut__h76225f74e93744a5 @ 82ae6b53e2f7420b55e4.wasm:0x6e5b37
f @ wasm.js:230
o @ wasm.js:215
Promise.then (async)
P._.wbg.__wbg_then_f7e06ee3c11698eb @ wasm.js:3232
$func1485 @ 82ae6b53e2f7420b55e4.wasm:0x61ca5c
$func2977 @ 82ae6b53e2f7420b55e4.wasm:0x6d095b
$wasm_bindgen__convert__closures__invoke2_mut__h0edc44cbd56288e2 @ 82ae6b53e2f7420b55e4.wasm:0x6e5a43
T @ wasm.js:321
(anonymous) @ wasm.js:3072
P._.wbg.__wbg_new_43f1b47c28813cbd @ wasm.js:3077
$programmanager_buildDeploymentTransaction @ 82ae6b53e2f7420b55e4.wasm:0x6d4df4
buildDeploymentTransaction @ wasm.js:1437
deploy @ program-manager.ts:189
await in deploy (async)
(anonymous) @ worker.js:321
h @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
l @ worker.js:2
s @ worker.js:2
Promise.then (async)
l @ worker.js:2
i @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:296
wasm.js:3027 Inserting externally provided fee proving and verifying keys
wasm.js:3027 Authorizing Fee
wasm.js:3027 Executing fee
wasm.js:3211 Fetch finished loading: GET "https://api.explorer.aleo.org/v1/testnet3/latest/stateRoot".
P._.wbg.__wbg_fetch_b5d6bebed1e6c2d2 @ wasm.js:3211
$func115 @ 82ae6b53e2f7420b55e4.wasm:0x11d67a
$func527 @ 82ae6b53e2f7420b55e4.wasm:0x3c72d2
$func157 @ 82ae6b53e2f7420b55e4.wasm:0x1d6777
$func175 @ 82ae6b53e2f7420b55e4.wasm:0x21913f
$func1806 @ 82ae6b53e2f7420b55e4.wasm:0x65a8d3
$func2665 @ 82ae6b53e2f7420b55e4.wasm:0x6ba70b
$wasm_bindgen__convert__closures__invoke1_mut__h76225f74e93744a5 @ 82ae6b53e2f7420b55e4.wasm:0x6e5b37
f @ wasm.js:230
o @ wasm.js:215
Promise.then (async)
P._.wbg.__wbg_then_f7e06ee3c11698eb @ wasm.js:3232
$func1485 @ 82ae6b53e2f7420b55e4.wasm:0x61ca5c
$func2977 @ 82ae6b53e2f7420b55e4.wasm:0x6d095b
$wasm_bindgen__convert__closures__invoke2_mut__h0edc44cbd56288e2 @ 82ae6b53e2f7420b55e4.wasm:0x6e5a43
T @ wasm.js:321
(anonymous) @ wasm.js:3072
P._.wbg.__wbg_new_43f1b47c28813cbd @ wasm.js:3077
$programmanager_buildDeploymentTransaction @ 82ae6b53e2f7420b55e4.wasm:0x6d4df4
buildDeploymentTransaction @ wasm.js:1437
deploy @ program-manager.ts:189
await in deploy (async)
(anonymous) @ worker.js:321
h @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
l @ worker.js:2
s @ worker.js:2
Promise.then (async)
l @ worker.js:2
i @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:2
(anonymous) @ worker.js:296
contentScript.js:1 Intercom connection corrupted A {message: 'Not Found', errors: undefined, name: 'IntercomError'}
wasm.js:3027 Verifying fee execution
wasm.js:3027 Verifying the deployment and fees
wasm.js:3027 Creating deployment transaction
worker.js:337 Error: Error posting transaction: No response received: Failed to execute 'fetch' on 'WorkerGlobalScope': Request with GET/HEAD method cannot have body.
    at u.submitTransaction (network-client.ts:610:13)
    at async p.deploy (program-manager.ts:190:9)
```

Reply:
That should be an easy fix.